### ISSUE TITLE
CSV: Server 2012 R2 preventing plugins loading

### ISSUE THREAD

**Issue by keithjbell** [NewSpring/Excavator#111]
I have been unable to run excavator to do our first import of data to Rock. It keeps crashing. 
![screenshot 2](https://cloud.githubusercontent.com/assets/9340628/5838224/80dd4908-a14b-11e4-92ba-5076f88ac0bc.png)

**Comment by dcs619** [comment]
That's actually a Windows permissions issue on the files.  Can you set
read/write folder permissions on the download and run it as administrator?

If the top bar is whited out, Windows is preventing the necessary plugins
from being loaded.

**Comment by keithjbell** [comment]
I'm still missing something I guess. I download the "master-zip" file. Inside that folder what am I suppose to set the permissions on? I did try to run the "excavator application" as administrator but had the same results.

**Comment by dcs619** [comment]
Can you right-click the unzipped folder and remove the Read-Only property?  Windows sets this property by default for any downloaded files.  
![screenshot 2015-01-21 10 19 17](https://cloud.githubusercontent.com/assets/1210933/5838691/1ad838d2-a157-11e4-8ef6-612fc79d2e2c.png)

You might also need to set Execute permissions to Everyone on the Security tab.

**Comment by keithjbell** [comment]
It's still happening. I did notice that I uncheck the "read-only" and hit apply, then rechecking it after I close the dialog box it is rechecked.

**Comment by dcs619** [comment]
What OS are you running?

David Stevens

*Web Developer*NewSpring Church
www.newspring.cc
864-965-9540

**Comment by keithjbell** [comment]
Server 2012 R2 Standard

**Comment by dcs619** [comment]
Ok.  I have a Server 2012 setup but I'll have to build a new version and test.

**Comment by keithjbell** [comment]
Ok. I didn't realize I was trying something this wasn't ready for. No worries. I've got a few weeks to get this import and training done. Keep me or this thread posted if possible. I don't want to hassle you unnecessarily. I appreciate your hard work. :) Thanks for the immediate responses

**Comment by dcs619** [comment]
No worries...Excavator maintenance is a nights/weekends thing right now so it's a challenge to keep up with the pace of Rock releases and test against each one.

**Comment by keithjbell** [comment]
Any release plan for this yet? Just checking in... Thanks for the extra time and work to make this click. :)

**Comment by dcs619** [comment]
@keithjbell I just pushed a new zip file, can you try it and see?  I wasn't able to reproduce the permissions issue on my server.

**Comment by dcs619** [comment]
@keithjbell Any further updates?

**Comment by keithjbell** [comment]
i tried very quickly the other day to do it but I may have missed a crucial step in opening up and launching it correctly. My plan is to try again tomorrow. I’m slammed with a couple video projects today. Thanks for your help and for checking in with me. 







Keith Bell. Creative Arts Minister

Office: 417.673.1245 - ext.18

Cell: 1.918.919.2076

Email: keith@cartervillechristian.com

Twitter: keithjbell

Blog: Real-Time Tech

Watch Sermons Online

**Comment by keithjbell** [comment]
I'm still having the same results. I followed the previous directions of taking "read only" permissions off of the downloaded folder. Any thoughts?

**Comment by dcs619** [comment]
Does it look the same as the above screenshot?

**Comment by keithjbell** [comment]
ok... I just tried a couple steps that I believe may have got me somewhere. I right clicked on the zipped folder and chose properties. Then there was an "unblock" option in the lower corner of the dialog box. I clicked on that. Extracted folder. Right clicked on "excavator" extracted folder, properties. Unchecked "read only", apply. Right clicked on "excavator" application, "run as administrator" and it seems to be running ok. I'll let you know if I get snagged somewhere else. Thanks!

**Comment by keithjbell** [comment]
I spoke to soon... It crashed as soon as I tried to "connect" to the database server. I'm going to keep messing with it for the next 30 minutes or so.

**Comment by dcs619** [comment]
Awesome!

The User Account Control (UAC) in Win7+ has been the biggest frustration
running Excavator on other systems.  I don't have a MS-signed security
certificate so I'm guessing that's why it's blocked.

**Comment by dcs619** [comment]
You can set the database connection via the .config file for now -- the
Connect button just makes it easier for non-technical users.

**Comment by keithjbell** [comment]
With skipping the "connect" button I successfully can navigate to my CSV however when I click "next" it crashes as well.

**Comment by keithjbell** [comment]
Im currently uploading a small video of me going through the steps. Hopefully it's just me doing something wrong. I'll send you a link shortly.

**Comment by keithjbell** [comment]
Check this out and let me know if I'm missing a step. https://dl.dropboxusercontent.com/u/2678595/excavator_rockissue.mp4

**Comment by dcs619** [comment]
It sounds like the SQL connector is still blocked.

**Comment by keithjbell** [comment]
Is that something that I've got wrong on my end?

**Comment by bperreault** [comment]
Hi, Keith
I wonder if your CSV file is formatted to the specifications.  On the video, after you upload your CSV you should see a message in Green that says Successfully Read the Import File.  I can't tell what it says, but the message is in yellow which means it was not successful.  But I'll see what else I can find.

**Comment by bperreault** [comment]
It looks like the source .csv file needs to be somewhere other than the desktop. Have you tried placing "people list data grid.csv" into the Documents folder?

**Comment by dcs619** [comment]
The source CSV needs to start with Individual or Family in the name to be
matched.  Also, if the file is open during the import it will be locked
meaning Excavator can't access it.

David Stevens

*Web Developer*NewSpring Church
www.newspring.cc
864-965-9540

**Comment by dcs619** [comment]
@keithjbell I watched the video and your CSV file needs to start with "Individual" or "Family" in the name so that Excavator can parse it.  Check the documentation for details: https://github.com/NewSpring/Excavator/wiki/3.-CSV-Import#IndividualInformation

That won't fix the DLL blocking that's going on but I'll try unpackaging them and see if that works.

**Comment by keithjbell** [comment]
I'm spending a couple hours this morning on this. I guess I'm missing where to find the specs for the csv are. I've wandered around your excavator github page and I'm not having any luck finding a readme or some other "formatting" requirements for the csv. Also, the yellow warning that comes up is "could not open the source file. Please make sure the file is not in use"

I don't have it open or anything so I'm not sure why I'm getting that message. I moved it to the documents folder and still am not having any luck. Thoughts?

**Comment by dcs619** [comment]
Documentation is on the wiki: https://github.com/NewSpring/Excavator/wiki/3.-CSV-Import

The biggest issue seemed like your source file has to start with "Individual" or "Family" in the name or Excavator will not recognize it.

**Comment by keithjbell** [comment]
Thank you!!! I'll check back in after a thorough check of all this.

**Comment by keithjbell** [comment]
ok. I've successfully formatted the csv's to spec. My problem now is connecting to the database server. Each time I click the "connect" button Excavator crashes. Thoughts?

**Comment by arranf** [comment]
Right click on all the files and click unblock.

**Comment by Sylhos** [comment]
Hello guys.  I am having a very similar issue, and I have not found a workaround for mine either.  My crash occurs even if I click the "upload" button (the "connect" crashes also).  I have checked the folder permissions and everything is read/write.  I even placed the code on a d: drive folder to avoid c: restrictions.  I am on an 64 bit version of 2012 R2 trying to import a FellowshipOne mdb file.  Any suggestions is much appreciated.

**Comment by arranf** [comment]
@Sylhos Just to double check (you didn't mention this in your post) have you clicked 'unblock' on the .exe and .dll files

**Comment by Sylhos** [comment]
@arranf  Sorry I missed that.  It worked after I unblocked each dll/exe's.  Now I am getting Houshold_ID does not exist message, but I believe is related to the MDF file.  Thanks!

**Comment by dcs619** [comment]
@Sylhos Did you mean Household_ID or Houshold_ID?  Houshold_ID is a misspelling but if that's the error you're getting that should be fixed in code.

**Comment by dcs619** [comment]
@keithjbell @Sylhos I did a fair amount of research this weekend + tried refactoring Excavator to get rid of this problem.  Unfortunately the blocking is part of security auditing introduced in Windows 7 and Server 2012.  You can read more about the issue [here](http://news.kynosarges.org/2015/01/30/stop-automatic-execution-blocking) and [here](http://weblogs.asp.net/dixin/understanding-the-internet-file-blocking-and-unblocking). 

The only thing I found that helps is to open the properties on Excavator.zip immediately after download -- clicking Unblock on the .zip **before** extracting keeps the extracted files from being blocked.

I'm closing this issue as there's nothing else I can do to prevent the files from blocking.  The SQL Connect button is a valid bug that I still need to fix [here](https://github.com/NewSpring/Excavator/issues/119).  However, you can set the database connection manually via Excavator.exe.config and start your import.
